src/Symfony/Bridge/Doctrine/PropertyInfo/DoctrineExtractor.php

```diff
@@ -294,7 +294,7 @@ private function getPhpType(string $doctrineType): ?string
             case Types::DATEINTERVAL:
                 return Type::BUILTIN_TYPE_OBJECT;
 
-            case Types::ARRAY:
+            case 'array':
```

Comment:
I suggest adding comments near those saying they are DBAL 3 types (`json_array` below should have it saying it is a DBAL 2 type). This would allow identifying those cases to clean them when dropping support for older DBAL versions.